src/components/intro/language-switcher/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { useRouter, usePathname } from "next/navigation";
+import { useEffect, useState } from "react";
+import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
+import { localeLiterals } from "../../../constants/local-literals";
+
+type Locale = "en" | "ru";
+
+const LocaleSwitcher = () => {
+  const router = useRouter();
+  const pathname = usePathname();
+  const [isMounted, setIsMounted] = useState(false);
+  const [currentLocale, setCurrentLocale] = useState<Locale>("en");
+
+  useEffect(() => {
+    setIsMounted(true);
+    const locale = pathname.split("/")[1] as Locale;
+    if (locale === "en" || locale === "ru") {
```

Comment:
I've implemented locale enums in the language switcher branch . However, I haven’t included these enums in the footer branch to avoid mixing changes at this stage. 

If needed, I can merge the language switcher branch into the footer branch